gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionCompletionContributor.java

```diff
@@ -29,15 +30,22 @@
 import com.intellij.psi.PsiElement;
 import com.intellij.psi.util.PsiTreeUtil;
 import com.intellij.util.ProcessingContext;
+import com.palantir.gradle.versions.intellij.VersionExplorer.GroupAndDep;
 import com.palantir.gradle.versions.intellij.psi.VersionPropsDependencyVersion;
 import com.palantir.gradle.versions.intellij.psi.VersionPropsProperty;
 import com.palantir.gradle.versions.intellij.psi.VersionPropsTypes;
-import java.util.stream.IntStream;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 import one.util.streamex.StreamEx;
 
 public class VersionCompletionContributor extends CompletionContributor {
-
-    private static final RepositoryExplorer repositoryExplorer = new RepositoryExplorer();
+    private static final GroupPartOrPackageNameExplorer GROUP_PART_OR_PACKAGE_NAME_EXPLORER =
+            new GroupPartOrPackageNameExplorer();
```

Comment:
This means there's two `GroupPartOrPackageNameExplorer` instances, one in this completer, one in the other. Can we make these `Explorer` classes services somehow so there's only ever one instance of them?

Reply:
converted to application services